tests/test_trusted_metadata_set.py

```diff
@@ -110,13 +128,269 @@ def test_update_with_invalid_json(self):
 
             # metadata is of wrong type
             with self.assertRaises(exceptions.RepositoryError):
-                update_func(data["root"])
+                update_func(self.metadata["root"])
```

Comment:
Based on the tests you wrote I guess this does not actually test what I thought it would -- for snapshot and targets this is going to fail in metafile verification already, not because the type is wrong ... feel free to remove this if you have the root/timestamp cases covered elsewhere

Reply:
This won't work only for snapshot where there are hashes and length verification.
I handle this case in one commit.